util/find-doc-nits

```diff
@@ -203,6 +205,153 @@ sub check_head_style {
     }
 }
 
+# Because we have options and symbols with extra markup, we need
+# to take that into account, so we need a regexp that extracts
+# markup chunks, including recursive markup.
+# please read up on /(?R)/ in perlre(1)
+# (note: order is important, (?R) needs to come before .)
+# (note: non-greedy is important, or something like 'B<foo> and B<bar>'
+# will be captured as one item)
+my $markup_re =
+    qr/(                        # Capture group
+           [BIL]<               # The start of what we recurse on
+           (?:(?-1)|.)*?        # recurse the whole regexp (refering to
+                                # the last opened capture group, i.e. the
+                                # start of this regexp), or pick next
+                                # character.  Do NOT be greedy!
+           >                    # The end of what we recurse on
+       )/x;                     # (the x allows this sort of split up regexp)
+
+# Options must start with a dash, followed by a letter, possibly
+# followed by letters, digits, dashes and underscores, and the last
+# character must be a letter or a digit.
+# We do also accept the single -? or -n, where n is a digit
+my $option_re =
+    qr/(?:
+            \?                  # Single question mark
+            |
+            \d                  # Single digit
+            |
+            [[:alpha:]](?:[-_[:alnum:]]*?[[:alnum:]])?
+       )/x;
+
+# Checks of command option (man1) formatting.  The man1 checks are
+# restricted to the SYNOPSIS and OPTIONS sections, the rest is too
+# free form, we simply cannot be too strict there.
+
+sub is_option_normalise {
+    my ($id, $filename, $option) = @_;
+
+    # $unwrapped is the option with the outer B<> markup removed
+    my $unwrapped = $option;
+    $unwrapped =~ s/^B<//;
+    $unwrapped =~ s/>$//;
+    # $unmarked is the option with *all* markup removed
+    (my $unmarked = $option) =~ s/[BIL]<|>//msg;
+
+    # If we found an option, check it, collect it
+    if ($unwrapped =~ /^\s*-/) {
+        if ($unmarked =~ /^-${option_re}$/) {
+            return $unwrapped;
+        } else {
+            return "";          # Malformed option
+        }
+    }
+    return undef;               # Something else
```

Comment:
should probably have a comment at the start of this function explaining what the three different return-types mean.

Reply:
Okie